### ISSUE TITLE
boxcar fails to build on PowerPC: no `AtomicU64` in `sync::atomic`

### ISSUE THREAD

**Issue by mgorny** [ibraheemdev/boxcar#19]
Noticed while trying to install [uv](https://github.com/astral-sh/uv/):

```
   Compiling boxcar v0.2.7
     Running `/opt/rust-bin-1.82.0/bin/rustc --crate-name boxcar --edition=2021 /var/tmp/portage/dev-python/uv-0.5.6/work/cargo_home/gentoo/boxcar-0.2.7/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 --check-cfg 'cfg(docsrs)' --check-cfg 'cfg(feature, values())' -C metadata=2d0b964ddf83b78b -C extra-filename=-2d0b964ddf83b78b --out-dir /var/tmp/portage/dev-python/uv-0.5.6/work/uv-0.5.6/target/powerpc-unknown-linux-gnu/release/deps --target powerpc-unknown-linux-gnu -C strip=symbols -L dependency=/var/tmp/portage/dev-python/uv-0.5.6/work/uv-0.5.6/target/powerpc-unknown-linux-gnu/release/deps -L dependency=/var/tmp/portage/dev-python/uv-0.5.6/work/uv-0.5.6/target/release/deps --cap-lints allow -C strip=none -C linker=powerpc-unknown-linux-gnu-gcc -C link-arg=-Wl,-O1 -C link-arg=-Wl,--as-needed`
error[E0432]: unresolved import `core::sync::atomic::AtomicU64`
 --> /var/tmp/portage/dev-python/uv-0.5.6/work/cargo_home/gentoo/boxcar-0.2.7/src/raw.rs:6:49
  |
6 | use core::sync::atomic::{AtomicBool, AtomicPtr, AtomicU64, AtomicUsize, Ordering};
  |                                                 ^^^^^^^^^
  |                                                 |
  |                                                 no `AtomicU64` in `sync::atomic`
  |                                                 help: a similar name exists in the module: `AtomicU32`
```

`core::sync::atomic::AtomicU64` is not portable. You need to use [portable-atomic](https://crates.io/crates/portable-atomic) crate instead. See e.g. https://github.com/edera-dev/tokio-tar/pull/1/files for an example how to do it.

**Comment by ibraheemdev** [comment]
Thanks for pointing this out. We can stick to `AtomicU32` on 32-bit platforms and use a CAS loop to avoid overflow.

**Comment by lscorcia** [comment]
Same issue here on armv5te-unknown-linux-gnueabi , also caused by uv dependency.

**Comment by ibraheemdev** [comment]
Released in 0.2.8.

**Comment by mgorny** [comment]
Thanks. I can confirm that it works now.
